### PR TITLE
feat: add support for session secrets in Renku 2.0

### LINKED ISSUE #3832: [ShapeUp] Session Secrets in Renku 2.0
[Session Secrets in Renku 2.0](https://renku.notion.site/Session-Secrets-in-Renku-2-0-b792b9a0f5b7498e9f2fc31b6ba61097)

### PR DESCRIPTION
**Pull request by leafty**
Closes #3832.

Add support for session secret slots and session secrets in Renku 2.0.

/deploy

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -12,9 +17,20 @@ Gitlab deployment as specified in our `documentation <https://renku.readthedocs.
 before installing this or any subsequent Renku version. Gitlab publishes an official Helm chart and
 that is what should be used for deploying Gitlab with Helm.
 
+User-Facing Changes
+~~~~~~~~~~~~~~~~~~~
+
+**🌟 New Features**
+
+- **UI**: Configure and save session secrets in Renku 2.0 projects and use them in sessions (TODO).
```
Comment (rokroskar) [inline CHANGELOG.rst:25]: Why `TODO`?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -3,6 +3,11 @@
 0.62.0
 ------
 
+Session secrets make it possible to connect to protected resources, such as databases or external
+compute systems, from a Renku session in a standardized and shareable manner. Collaborating with
+secrets is easy too: configure a single session secret slot to ensure that the secret shows up
+the same way for everyone, and each person enters their own value.
```
Comment (rokroskar) [inline CHANGELOG.rst:9]: Do we need to make it clearer that the secrets themselves are not shared, just the configuration?

File: helm-chart/renku/templates/network-policies.yaml [hunk 3]
```diff
@@ -991,6 +991,33 @@ spec:
 ---
 apiVersion: networking.k8s.io/v1
 kind: NetworkPolicy
+metadata:
+  # Needed for secret mounting
+  name: ingress-to-data-service-from-v2-sessions
```
Comment (rokroskar) [inline helm-chart/renku/templates/network-policies.yaml:996]: This means the data service can be access on a cluster IP from within any user session?

File: CHANGELOG.rst (hunk 1 above)
Comment (olevski) [inline CHANGELOG.rst:25]: I think that is a leftover from editing?

```suggestion
- **UI**: Configure and save session secrets in Renku 2.0 projects and use them in sessions.
```

**Review by rokroskar — COMMENTED** [review]
Thanks @leafty! just a couple of comments

File: CHANGELOG.rst (hunk 1 above)
Comment (leafty) [reply CHANGELOG.rst]: Fixed in c023be3629f0d063da801d9f582fe3a8b9f39a47.

File: helm-chart/renku/templates/network-policies.yaml (hunk 3 above)
Comment (leafty) [reply helm-chart/renku/templates/network-policies.yaml:996]: Yes, it is already the case for v1 sessions.

File: CHANGELOG.rst (hunk 1 above)
Comment (leafty) [reply CHANGELOG.rst]: Fixed in c023be3629f0d063da801d9f582fe3a8b9f39a47.

File: CHANGELOG.rst (hunk 2 above)
Comment (leafty) [reply CHANGELOG.rst:9]: tagging @lokijuhy who wrote the release text.

Comment (leafty) [reply CHANGELOG.rst]: Oopsie, I had the old copy. Now updated with 5685b710ceb6740d2b7377145ff48a133b91b918.

**Review by lokijuhy — APPROVED** [review]

**Review by lokijuhy — COMMENTED** [review]
Are all of @andre-code's changes supposed to be included in the changelog here? I see them mentioned in the [UI tagged release](https://github.com/SwissDataScienceCenter/renku-ui/releases/tag/3.43.0).

**Review by olevski — APPROVED** [review]

**Comment by leafty** [comment]
> Are all of @andre-code's changes supposed to be included in the changelog here? I see them mentioned in the [UI tagged release](https://github.com/SwissDataScienceCenter/renku-ui/releases/tag/3.43.0).

Fixed in 5a48991d81743732750467517b3eb16b15ac2afa.

**Pull request merged**